tls/s2n_server_new_session_ticket.c

```diff
@@ -138,7 +138,9 @@ S2N_RESULT s2n_tls13_server_nst_send(struct s2n_connection *conn, s2n_blocked_st
     RESULT_GUARD_POSIX(s2n_stuffer_growable_alloc(&nst_stuffer, maximum_nst_size));
```

Comment:
Nit: would be nice to avoid allocation if we have no keys to encrypt anyway (:

Reply:
Lindsay is working on some PRs that should make our memory use smaller, I think those will be more impactful than conditioning this allocation.